lib/us_core_test_kit/search_test.rb

```diff
@@ -568,7 +568,24 @@ def search_param_value(name, patient_id, include_system: false)
           when FHIR::Address
             element.text || element.city || element.state || element.postalCode || element.country
           else
-            element
+            if metadata.version != 'v3.1.1' &&
+               metadata.search_definitions[name.to_sym][:type] == 'date' &&
+               params_with_comparators&.include?(name)
+              # convert date search to greath-than comparator search with correct precision
+              # For all date search parameters:
+              #   Patient.birthDate does not mandate comparators so cannot be converted
+              #   Goal.target-date has day precision
+              #   All others have second + time offset precision
+              if /^\d{4}$/.match?(element) || # YYYY
+                 /^\d{4}-\d{2}$/.match?(element) || # YYYY-MM
+                (/^\d{4}-\d{2}-\d{2}$/.match?(element) && resource_type != "Goal") # YYYY-MM-DD AND Resource is NOT Goal
+                "gt#{(DateTime.xmlschema(element)-1).xmlschema}"
```

Comment:
```suggestion
                "gt#{(DateTime.xmlschema(element)-1).xmlschema}"
```
Could you also explain the logic behind subtracting 1? I'm assuming it has to do with subtracting by some amount so that the greater than comparator returns what you'd expect, but I want to confirm.

Reply:
The logic is to convert year/month/day to range compare to cover the correct date time range to ensure that search has at least one result

2020 => gt2019-12-31T00:000:00 # so the resource with date value '2020' is in the result

2020-04 => gt2020-03-31T00:00:00

2020-04-12 => gt2020-04-11T00:00:00

Reply:
> Code looks good, but I guess I'm a little fuzzy on the logic here. For datatype date, why are you adding precision to second + time offset? I read the guidance as saying datatype date only needs be precise to the day.

That is a good point. I will fix that.

Reply:
We use time+offset in comparator search for both the date and date-time precision. After some investigation, I think our solution is OK.
1) time+offset is a valid presentation of date search parameter, which is defined as "The date parameter format is yyyy-mm-ddThh:mm:ss[Z|(+|-)hh:mm] (the standard XML format)."
2) comparator + time-offset presents a valid and meaning range search. For example: gt2022-05-14T00:00:00+00:00 is a valid range without any ambiguity no matter for date or datetime target value.